sources/packages/api/src/route-controllers/application/application.controller.ts

```diff
@@ -86,13 +88,26 @@ export class ApplicationController extends BaseController {
         payload.associatedFiles,
       );
     }
-
+    // TODO:remove the static program year and add dynamic year, from program year table
+    const sequenceName = "2122";
+    let nextApplicationSequence: number = NaN;
+    await this.sequenceService.consumeNextSequence(
+      sequenceName,
+      async (nextSequenceNumber: number) => {
+        nextApplicationSequence = nextSequenceNumber;
+      },
+    );
+    if (!nextApplicationSequence) {
+      throw new BadRequestException();
+    }
+    const applicationNumber =
+      parseInt(sequenceName) * 10 ** 6 + (nextApplicationSequence - 1);
```

Comment:
Please consider that the sequenceName could be a string. I do not see a benefit considering that it will be a number.

Reply:
I believe that the logic could be simplified as below:
```
const MAX_APPLICATION_NUMBER_LENGTH = 10;
const sequenceNumberSize = MAX_APPLICATION_NUMBER_LENGTH - sequenceName.length;
const applicationNumber = sequenceName + nextApplicationSequence.padStart(sequenceNumberSize, "0") ;
```